src/accountsdb/index.zig

```diff
@@ -925,11 +925,15 @@ pub const DiskMemoryAllocator = struct {
     count: usize = 0,
     mux: std.Thread.Mutex = .{},
 
+    /// HashMap that maps the Mmap-ed memory address of a file to the file's index number
+    hashmap: std.AutoHashMap([*]u8, usize),
```

Comment:
I'm not a huge fan of the fact that this introduces a dependency on another allocator which has to potentially also be thread safe. I would like to instead see an approach where the metadata is placed in (disk) memory before or after the user-facing buffer.